### ISSUE TITLE
Unhelpful error when columns are not matched in informant-based pipeline

### ISSUE THREAD

**Issue by petrbouchal** [rstudio/pointblank#344]
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) and [contributing guidelines](https://github.com/rich-iannone/pointblank/blob/master/.github/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/rich-iannone/pointblank/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [ x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

In an informant workflow, when the selector in `info_columns()` does not match any columns:
- an error is produced if another `info_columns()` is appended to the pipe (Example 1 below)
- an `NA` column is reported in the output if the `info_columns()` call is at the end of the pipeline (Example 2 below)

This might be driven by the same internal issue as #275.

## Reproducible example

* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

### Example 1

``` r
library(pointblank)

mtcars %>% 
  create_informant() %>% 
  info_columns(matches("x"), info = "hi") %>% 
  info_columns(matches("mpg"), info = "there")
```

    #> Error: Column 12 must be named.
    #> Use .name_repair to specify repair.

<sup>Created on 2021-08-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

### Example 2:

``` r
library(pointblank)

mtcars %>% 
  create_informant() %>% 
  info_columns(matches("x"), info = "hi")
```

- runs fine (HTML output omitted from reprex output)
- table output lists all columns in `mtcars`
- table output contains row for column named `NA` with no information in the other field

## Expected result

For both scenarios, I would expect a warning or message and for the output to be unaffected by any `info_columns()` whose selectors match no columns. Alternatively, there might be some way to report in the output the failures to match columns.

## Session info

End the reproducible example with a call to `sessionInfo()` in the same session (e.g. `reprex(si = TRUE)`) and include the output.

**Comment by rich-iannone** [comment]
Thank you for this detailed report. Definitely something wrong here so a fix will come soon.

**Comment by rich-iannone** [comment]
@petrbouchal This is now fixed and I hope the fix goes far enough (it yields a warning with a message). Let me know if this is adequate or could use more work.

**Comment by petrbouchal** [comment]
Great, now as expected. Thank you for the rapid fix!

**Comment by petrbouchal** [comment]
Looking at this in use, I suspect it might be helpful to include the selection expression in the warning if it is not too difficult and wouldn't slow things down too much. With a longer pipeline of `info_columns()` calls, it can get difficult to identify which step the warning came from - especially in non-interactive situations.

**Comment by rich-iannone** [comment]
@petrbouchal I totally agree with this. Thanks for testing this out and making this recommendation, it will be implemented shortly.
